internal/reconciler/telemetry/metric_components_checker.go

```diff
@@ -109,3 +120,13 @@ func (m *metricComponentsChecker) addReasonPrefix(reason string) string {
 	}
 	return reason
 }
+
+func (m *metricComponentsChecker) checkTLSCertificateMessage(pipelines []telemetryv1alpha1.MetricPipeline) string {
```

Comment:
check implies that a boolean is to be returned
```suggestion
func (m *metricComponentsChecker) firstTLSCertificateMessage(pipelines []telemetryv1alpha1.MetricPipeline) string {
```